config/systems/todi.py

```diff
@@ -6,69 +6,102 @@
-            'descr': 'todi vcluster',
-            'hostnames': ['todi'],
-            'modules_system': 'nomod',
-            'partitions': [
+            'name': 'normal',
+            'descr': 'TODI Grace-Hopper GH200',
+            'scheduler': 'slurm',
+            'time_limit': '10m',
+            'container_platforms': [
+#                 {
+#                     'type': 'Sarus',
+#                     #'modules': ['sarus']
+#                 },
+#                 {
+#                     'type': 'Singularity',
+#                     #'modules': ['singularity/3.6.4-todi']
+#                 }
+            ],
+            'environs': [
+                'builtin',
+                'PrgEnv-cray',
+                'PrgEnv-gnu',
+            ],
+            'max_jobs': 100,
+            'extras': {
+                'cn_memory': 825,
+            },
+            'features': ['ce', 'gpu', 'nvgpu', 'remote', 'scontrol', 'uenv'],
+            'resources': [
+                {
+                    'name': 'switches',
+                    'options': ['--switches={num_switches}']
+                },
                 {
-                    'name': 'login',
-                    'scheduler': 'local',
-                    'time_limit': '10m',
-                    'environs': [
-                        'builtin',
-                    ],
-                    'descr': 'Login nodes',
-                    'max_jobs': 4,
-                    'launcher': 'local'
+                    'name': 'gres',
+                    'options': ['--gres={gres}']
                 },
                 {
-                    'name': 'normal',
-                    'scheduler': 'slurm',
-                    'time_limit': '10m',
-                    'environs': [
-                        'builtin',
-                    ],
-                    'max_jobs': 100,
-                    'extras': {
-                        'cn_memory': 500,
-                    },
-                    'resources': [
-                        {
-                            'name': 'memory',
-                            'options': ['--mem={mem_per_node}']
-                        },
-                    ],
-                    'features': ['ce', 'gpu', 'nvgpu', 'remote', 'scontrol', 'uenv'],
-                    'devices': [
-                        {
-                            'type': 'gpu',
-                            'arch': 'sm_90',
-                            'num_devices': 4
-                        }
-                    ],
-                    'launcher': 'srun'
+                    'name': 'memory',
+                    'options': ['--mem={mem_per_node}']
                 },
-            ]
+            ],
+            'devices': [
+                {
+                    'type': 'gpu',
+                    'arch': 'sm_90',
+                    'num_devices': 4
+                }
+                ],
+            'launcher': 'srun',
         },
+    ]
+}
+
+todi_sys = copy.deepcopy(base_config)
```

Comment:
The deepcopy makes sense only when there are two or more similar systems with small differences (like daint and dom). Will todi have a tds? otherwise you can just set the configuration directly, right?

Reply:
@teojgo ?

Reply:
For the moment you can leave it without a deepcopy. Once we have a better picture of all the clusters we can extract the base config and do it like this

Reply:
https://github.com/eth-cscs/cscs-reframe-tests/pull/220/commits/83bfb0583069c9734366e0188f7bf48daa585534